apps/web/src/routes/AuthLayoutRouter/components/ContextMenu/ContextMenu.jsx

```diff
@@ -84,6 +90,214 @@ export default function ContextMenu (props) {
+
+      // Remove from URL once detected
+      const newParams = new URLSearchParams(searchParams)
+      newParams.delete('newWidgetId')
+      navigate(`${location.pathname}?${newParams.toString()}`, { replace: true })
+    } else {
+      // Check localStorage as backup
+      const storedWidgetId = localStorage.getItem('hylo:last-created-widget')
+      const timestamp = parseInt(localStorage.getItem('hylo:last-created-timestamp') || '0')
+
+      // Only use stored ID if it's recent (within last 5 seconds)
+      if (storedWidgetId && Date.now() - timestamp < 5000) {
+        console.log('Found new widget ID in localStorage:', storedWidgetId)
+        setNewViewId(storedWidgetId)
+        localStorage.removeItem('hylo:last-created-widget')
+        localStorage.removeItem('hylo:last-created-timestamp')
+      }
+    }
+  }, [searchParams, location, navigate])
+
+  // Listen for custom events
+  useEffect(() => {
+    const handleNewWidgetEvent = (event) => {
+      if (event.detail?.widgetId) {
+        console.log('Received new widget event:', event.detail.widgetId)
+        setNewViewId(event.detail.widgetId)
+      }
+    }
+
+    window.addEventListener('hylo:new-widget-added', handleNewWidgetEvent)
+    return () => {
+      window.removeEventListener('hylo:new-widget-added', handleNewWidgetEvent)
+    }
+  }, [])
+
+  // Gather all child items from all widgets to detect new views
+  const allItems = useMemo(() => {
+    let items = []
+    // Collect items from child widgets
+    contextWidgets.forEach(widget => {
+      if (widget.childWidgets && widget.childWidgets.length > 0) {
+        items = items.concat(widget.childWidgets.map(child => child.id))
+      }
+    })
+    return items
+  }, [contextWidgets])
+
+  // Check for new views (including child views)
+  useEffect(() => {
+    if (previousChildItems.length > 0) {
+      const newItem = allItems.find(id => previousChildItems.indexOf(id) === -1)
+      if (newItem) {
+        setNewViewId(newItem)
+      }
+    }
+    previousChildItems = [...allItems]
+  }, [allItems])
+
+  /**
+   * Effect to handle scrolling to and highlighting newly added widgets or views
+   *
+   * Waits for new elements to appear in the DOM, then scrolls precisely to their position
+   * while applying visual animations (faster bounce + border flash) for feedback.
+   */
+  useEffect(() => {
+    // Target element could be either a new top-level widget or a new child view
+    const elementId = newWidget ? newWidget.id : newViewId
+    if (!elementId) return
+
+    // Initial delay to let React finish rendering
+    const initialTimer = setTimeout(() => {
+      // Function to find, reveal, and scroll to the target element
+      const findAndScrollToElement = (retryCount = 0) => {
+        // Try multiple selectors to find the element
+        const targetElement = newWidget
+          ? newWidgetRef.current
+          : document.querySelector(`[data-view-id="${elementId}"]`) ||
```

Comment:
huh? why cant there just be 1 way to find a widget?

Reply:
and why wouldnt it be just the widgetRef?

Reply:
it looks like this could grab a div or an li as a go down, shouldnt we only be putting this on one or the other of those elements?